core/src/main/scala/chisel3/internal/MonoConnect.scala

```diff
@@ -338,24 +339,24 @@ private[chisel3] object MonoConnect {
   def canBeSink(data:   Data, context_mod: RawModule): Boolean = traceFlow(true, data, context_mod)
   def canBeSource(data: Data, context_mod: RawModule): Boolean = traceFlow(false, data, context_mod)
 
-  /** Check whether two aggregates can be bulk connected (<=) in FIRRTL. (MonoConnect case)
+  /** Check whether two Data can be bulk connected (<=) in FIRRTL. (MonoConnect case)
     *
     * Mono-directional bulk connects only work if all signals of the sink are unidirectional
     * In the case of a sink aggregate with bidirectional signals, e.g. `Decoupled`,
-    * a `BiConnect` is necessary.
+    * a `BiConnect` (`chisel3.<>` or `chisel.:<>=`) is necessary.
     */
-  private[chisel3] def canBulkConnectAggregates(
-    sink:                  Aggregate,
-    source:                Aggregate,
+  private[chisel3] def canFirrtlConnectData(
+    sink:                  Data,
+    source:                Data,
     sourceInfo:            SourceInfo,
     connectCompileOptions: CompileOptions,
     context_mod:           RawModule
   ): Boolean = {
-    // Assuming we're using a <>, check if a bulk connect is valid in that case
+    // Assuming we're using a <>, check if a FIRRTL.<= connection operator is valid in that case
     def biConnectCheck =
-      BiConnect.canBulkConnectAggregates(sink, source, sourceInfo, connectCompileOptions, context_mod)
+      BiConnect.canFirrtlConnectData(sink, source, sourceInfo, connectCompileOptions, context_mod)
 
-    // Check that the Aggregate can be driven (not bidirectional or an input) to match Chisel semantics
+    // Check that the sink Data can be driven (not bidirectional or an input) to match Chisel semantics
```

Comment:
Isn't this basically wrong shouldn't this care about the context of the module it is in?

Reply:
Yes, but it's a bug best solved in a separate PR.